dashboard/app/helpers/users_helper.rb

```diff
@@ -245,15 +240,66 @@ def user_summary(user)
         pages_completed.each_with_index do |result, index|
           progress["#{level_id}_#{index}"] = {
             result: result,
-            submitted: submitted ? true : nil,
-            readonly_answers: readonly_answers ? true : nil
+            submitted: level_progress[:submitted],
+            readonly_answers: level_progress[:readonly_answers]
           }.compact
         end
       end
     end
     progress
   end
 
+  # Summarizes a user's level progress for bubble choice level (parent level and sublevels)
+  private def get_bubble_choice_progress(level, user, user_levels_by_level, script_level, paired_user_levels, include_timestamp)
+    progress = {}
+
+    best_sublevel_id = level.best_result_sublevel(user)&.id
+    return progress if best_sublevel_id.nil?
+
+    sum_time_spent = 0
+
+    # get progress for sublevels to save in levels hash
+    level.sublevels.each do |sublevel|
+      ul = user_levels_by_level.try(:[], sublevel.id)
+      sublevel_progress = get_level_progress(ul, script_level, paired_user_levels, include_timestamp)
+      progress[sublevel.id] = sublevel_progress
+      sum_time_spent += sublevel_progress[:time_spent] || 0
+    end
+
+    progress[level.id] = progress[best_sublevel_id].clone
+    progress[level.id][:time_spent] = sum_time_spent if sum_time_spent > 0
+    progress
+  end
+
+  # Summarizes a user's progress on a particular level
+  private def get_level_progress(user_level, script_level, paired_user_levels, include_timestamp)
+    completion_status = activity_css_class(user_level)
+    locked = user_level.try(:show_as_locked?, script_level.lesson) || script_level.lesson.lockable? && !user_level
+
+    # if the level has not been tried, less progress data is necessary
+    if completion_status == LEVEL_STATUS.not_tried
+      return {
+        status: completion_status,
+        locked: locked || nil
+      }
+    end
+
+    # a user_level is submitted if the state is submitted UNLESS it is a peer reviewable level that has been reviewed
+    submitted = !!user_level.try(:submitted) &&
+      !(user_level.level.try(:peer_reviewable?) && [ActivityConstants::REVIEW_REJECTED_RESULT, ActivityConstants::REVIEW_ACCEPTED_RESULT].include?(user_level.best_result))
+
+    return {
+      status: completion_status,
+      locked: locked || nil,
+      result: user_level.try(:best_result) || 0,
+      submitted: submitted || nil,
+      readonly_answers: !!user_level.try(:readonly_answers) || nil,
+      paired: (paired_user_levels.include? user_level.try(:id)) || nil,
+      last_progress_at: include_timestamp ? user_level&.updated_at&.to_i : nil,
+      time_spent: user_level&.time_spent&.to_i
```

Comment:
I actually had to look this up. So it sounds like the main difference is that using try, if a method is called that doesn't exist for the object then it will swallow the `NoMethodError` and return nil. I think moving towards `&` seems like the right call here. I'll update.

Reply:
TIL! thanks for that distinction.

Reply:
Okay so turns out Free Response levels have a `peer_reviewable` method but other types of levels don't so that's a valid use of `try` that I'll need to leave there.